core/services/periodicbackup/backup.go

```diff
@@ -0,0 +1,139 @@
+package periodicbackup
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+var (
+	fileName           = "db_backup.tar.gz"
+	minBackupFrequency = time.Minute
+)
+
+type backupResult struct {
+	size int64
+	path string
+}
+
+type (
+	DatabaseBackup interface {
+		Start() error
+		Close() error
+		RunBackupGracefully()
+	}
+
+	databaseBackup struct {
+		logger          *logger.Logger
+		databaseURL     url.URL
+		frequency       time.Duration
+		outputParentDir string
+		done            chan bool
+	}
+)
+
+func NewDatabaseBackup(frequency time.Duration, databaseURL url.URL, outputParentDir string, logger *logger.Logger) DatabaseBackup {
+	return &databaseBackup{
+		logger,
+		databaseURL,
+		frequency,
+		outputParentDir,
+		make(chan bool),
+	}
+}
+
+func (backup databaseBackup) Start() error {
+
+	ticker := time.NewTicker(backup.frequency)
+
+	go func() {
+		for {
+			select {
+			case <-backup.done:
+				ticker.Stop()
+				return
+			case <-ticker.C:
+				if backup.frequencyIsTooSmall() {
+					logger.Errorf("Database backup frequency (%s=%v) is too small. Please set it to at least %s", "DATABASE_BACKUP_FREQUENCY", backup.frequency, minBackupFrequency)
+					continue
```

Comment:
Wondering if this should error out on startup? Seems dangerous to proceed without actually backing up based on a misconfiguration. On the other hand, if there's proper alerting setup this should become known very quickly, but we don't have a guarantee that's set up. I'd vote to return the error.

Reply:
Agree it should bomb out if this hits

Reply:
yeah makes sense 👍🏼